recurring_contract/models/move.py

```diff
@@ -84,124 +84,49 @@ def action_invoice_re_open(self):
     def reconcile_after_clean(self):
         """
         Called after clean invoices. If invoices can be reconciled
-        with open payment, this will split the payment into three amounts :
-            - amount for reconciling the past invoices
-            - amount for reconciling the future invoices
-            - leftover amount that will stay in the client balance
-        Then the invoices will be reconciled again
-
-        Invoices should be opened or canceled. if they are canceled they will
-        first be reopened
+        with open payment, this will do it.
+        Invoices should be open when called.
         :return: True
         """
-        # At first we open again the cancelled invoices
-        cancel_invoices = self.filtered(lambda i: i.state == 'cancel')
-        cancel_invoices.button_draft()
-        cancel_invoices.action_post()
         today = date.today()
+        mvl_obj = self.env['account.move.line']
         for partner in self.mapped('partner_id'):
             invoices = self.filtered(lambda i: i.partner_id == partner)
-            past_invoices = invoices.filtered(lambda i: i.invoice_date <= today)
-            future_invoices = invoices - past_invoices
+            past_invoices = invoices.filtered(lambda i: i.date <= today)
```

Comment:
```suggestion
            past_invoices = invoices.filtered(lambda i: i.invoice_date_due <= today)
```
The due_date  defines when the invoice shoud be paid for so i think that's a better check to know if an invoice is in the past. Moreover, it's more explicit than date field.